### PR TITLE
Add table listing available blocks

### PR DESCRIPTION
**Pull request by tpdorsey**
Adds tables listing "built-in" blocks and blocks available via Collections.

Some minor changes to text to accommodate, including adding a screenshot of the Blocks UI.

### Example

[Preview](https://deploy-preview-8443--prefect-orion.netlify.app/concepts/blocks/)

blocks

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [ ] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`

### CODE REVIEW

File: docs/concepts/blocks.md [hunk 1]
```diff
@@ -13,11 +13,113 @@ tags:
 
 Blocks are a primitive within Prefect that enable the storage of configuration and provide an interface for interacting with external systems.
 
-Blocks are useful for configuration that needs to be shared across flow runs and between flows. For configuration that will change between flow runs, we recommend using [parameters](/concepts/flows/#parameters).
-
-With blocks, you are able to securely store credentials for authenticating with services like AWS, GitHub, Slack, or any other system you'd like to orchestrate with Prefect. Blocks also expose methods that provide pre-built functionality for performing actions against an external system. Blocks can be used to download data from or upload data to an S3 bucket, query data from or write data to a database, or send a message to a Slack channel.
-
-Blocks can also be created by anyone and shared with the community. You'll find blocks that are available for consumption in many of the published [Prefect Collections](/collections/catalog).
+With blocks, you are able to securely store credentials for authenticating with services like AWS, GitHub, Slack, or any other system you'd like to orchestrate with Prefect.
```
Comment (billpalombi) [inline docs/concepts/blocks.md:16]:
```suggestion
With blocks, you can securely store credentials for authenticating with services like AWS, GitHub, Slack, and any other system you'd like to orchestrate with Prefect. 
```

File: docs/concepts/blocks.md [hunk 2]
```diff
@@ -13,11 +13,113 @@ tags:
...
-
-With blocks, you are able to securely store credentials for authenticating with services like AWS, GitHub, Slack, or any other system you'd like to orchestrate with Prefect. Blocks also expose methods that provide pre-built functionality for performing actions against an external system. Blocks can be used to download data from or upload data to an S3 bucket, query data from or write data to a database, or send a message to a Slack channel.
-
-Blocks can also be created by anyone and shared with the community. You'll find blocks that are available for consumption in many of the published [Prefect Collections](/collections/catalog).
+With blocks, you are able to securely store credentials for authenticating with services like AWS, GitHub, Slack, or any other system you'd like to orchestrate with Prefect. 
+
+Blocks also expose methods that provide pre-built functionality for performing actions against an external system. Blocks can be used to download data from or upload data to an S3 bucket, query data from or write data to a database, or send a message to a Slack channel.
```
Comment (billpalombi) [inline docs/concepts/blocks.md:18]:
```suggestion
Blocks expose methods that provide pre-built functionality for performing actions against an external system. They can be used to download data from or upload data to an S3 bucket, query data from or write data to a database, or send a message to a Slack channel.
```

File: docs/concepts/blocks.md [hunk 3]
```diff
@@ -13,11 +13,113 @@ tags:
...
+Prefect Cloud and the Prefect server UI display a library of block types available for you to configure blocks that may be used by your flows.
+
+![Viewing the new block library in the Prefect UI](../img/ui/block-library.png)
+
+!!! tip "Blocks and parameters"
+    Blocks are useful for configuration that needs to be shared across flow runs and between flows.
+
+    For configuration that will change between flow runs, we recommend using [parameters](/concepts/flows/#parameters).
+
+## Prefect built-in blocks
```
Comment (billpalombi) [inline docs/concepts/blocks.md:35]: Are we pulling these from somewhere or will this be copy & paste to maintain?

**Review by billpalombi — COMMENTED** [review]
A couple of nits and a question

File: docs/concepts/blocks.md (hunk 3 above)
Comment (tpdorsey) [reply docs/concepts/blocks.md:35]: At the moment this is semi-automated, but does require some hand work. Keeping an eye on making it fully automated in future.

Comment (tpdorsey) [reply docs/concepts/blocks.md:35]: To be more specific, the majority of the table is pulled directly from JSON in our repos, but the JSON does not yet include some details such as documentation URLs for links. We'll work on filling that our over time to enable build-time automation for the entire table.

File: docs/concepts/blocks.md [hunk 4]
```diff
@@ -13,11 +13,113 @@ tags:
...
-Blocks can also be created by anyone and shared with the community. You'll find blocks that are available for consumption in many of the published [Prefect Collections](/collections/catalog).
+With blocks, you can securely store credentials for authenticating with services like AWS, GitHub, Slack, and any other system you'd like to orchestrate with Prefect. 
+
+Blocks expose methods that provide pre-built functionality for performing actions against an external system. They can be used to download data from or upload data to an S3 bucket, query data from or write data to a database, or send a message to a Slack channel.
+
+You may configure blocks through code or via the Prefect Cloud and the Prefect server UI.
+
+You can access blocks for both configuring flow [deployments](/concepts/deployments/) and directly from within your flow code.
+
+Prefect provides some built-in block types that you can use right out of the box. Additional blocks are available through [Prefect Collections](/collections/catalog/) &mdash to use these blocks you can `pip install` the collection, then register the blocks you want to use with Prefect Cloud or a Prefect server.
```
Comment (serinamarie) [inline docs/concepts/blocks.md:24]:
```suggestion
Prefect provides some built-in block types that you can use right out of the box. Additional blocks are available through [Prefect Collections](/collections/catalog/). To use these blocks you can `pip install` the collection, then register the blocks you want to use with Prefect Cloud or a Prefect server.
```

File: docs/concepts/blocks.md [hunk 5]
```diff
@@ -13,11 +13,113 @@ tags:
...
+    For configuration that will change between flow runs, we recommend using [parameters](/concepts/flows/#parameters).
+
+## Prefect built-in blocks
+
+Prefect provides a broad range of commonly used, built-in block types. These block types are available in Prefect Cloud and the Prefect server UI and local installations of Prefect.
+
+| Block | Slug | Description |
+| --- | --- | --- |
+| [Azure](/concepts/filesystems/#azure) | `azure` | Store data as a file on Azure Datalake and Azure Blob Storage. |
+| [Date Time](/api-ref/prefect/blocks/system/#prefect.blocks.system.DateTime) | `date-time` | A block that represents a datetime |
```
Comment (serinamarie) [inline docs/concepts/blocks.md:42]:
```suggestion
| [Date Time](/api-ref/prefect/blocks/system/#prefect.blocks.system.DateTime) | `date-time` | A block that represents a datetime. |
```

File: docs/concepts/blocks.md [hunk 6]
```diff
@@ -13,11 +13,113 @@ tags:
...
+| [Pager Duty Webhook](/api-ref/prefect/blocks/notifications/#prefect.blocks.notifications.PagerDutyWebHook) | `pager-duty-webhook` | Enables sending notifications via a provided PagerDuty webhook. |
+| [Process](/concepts/infrastructure/#process) | `process` | Run a command in a new process. |
+| [Remote File System](/concepts/filesystems/#remote-file-system) | `remote-file-system` | Store data as a file on a remote file system.  Supports any remote file system supported by `fsspec`.  |
+| [S3](/concepts/filesystems/#s3) | `s3` | Store data as a file on AWS S3. |
+| [Secret](/api-ref/prefect/blocks/system/#prefect.blocks.system.Secret) | `secret` | A block that represents a secret value. The value stored in this block will be obfuscated when this block is logged or shown in the UI. |
+| [Slack Webhook](/api-ref/prefect/blocks/notifications/#prefect.blocks.notifications.SlackWebhook) | `slack-webhook` | Enables sending notifications via a provided Slack webhook. |
+| [SMB](/concepts/filesystems/#smb) | `smb` | Store data as a file on a SMB share. |
+| [String](/api-ref/prefect/blocks/system/#prefect.blocks.system.String) | `string` | A block that represents a string |
```
Comment (serinamarie) [inline docs/concepts/blocks.md:60]:
```suggestion
| [String](/api-ref/prefect/blocks/system/#prefect.blocks.system.String) | `string` | A block that represents a string. |
```

**Review by serinamarie — COMMENTED** [review]
LGTM 🙂

File: docs/concepts/blocks.md (hunk 3 above)
Comment (desertaxle) [reply docs/concepts/blocks.md:35]: The integrations team can work on making sure that all blocks have a documentation URL and resolve any other missing metadata. If this table would benefit from a custom view for blocks (like with relative URLs for the blocks that are in the core library) let us know and we can work on building that.

File: docs/concepts/blocks.md [hunk 7]
```diff
@@ -13,11 +13,113 @@ tags:
...
+| [Kubernetes Job](/api-ref/prefect/infrastructure/#prefect.infrastructure.KubernetesJob) | `kubernetes-job` | Runs a command as a Kubernetes Job. |
+| [Local File System](/concepts/filesystems/#local-filesystem) | `local-file-system` | Store data as a file on a local file system. |
+| [Microsoft Teams Webhook](/api-ref/prefect/blocks/notifications/#prefect.blocks.notifications.MicrosoftTeamsWebhook) | `ms-teams-webhook` | Enables sending notifications via a provided Microsoft Teams webhook. |
+| [Opsgenie Webhook](/api-ref/prefect/blocks/notifications/#prefect.blocks.notifications.OpsgenieWebhook) | `opsgenie-webhook` | Enables sending notifications via a provided Opsgenie webhook. |
+| [Pager Duty Webhook](/api-ref/prefect/blocks/notifications/#prefect.blocks.notifications.PagerDutyWebHook) | `pager-duty-webhook` | Enables sending notifications via a provided PagerDuty webhook. |
+| [Process](/concepts/infrastructure/#process) | `process` | Run a command in a new process. |
```
Comment (discdiver) [inline docs/concepts/blocks.md:54]: This matches elsewhere in the docs - but I feel like the key thing with Process is the customizability.

**Review by discdiver — APPROVED** [review]
This is great @tpdorsey! Thank you. Just minor nits and questions.

File: docs/concepts/blocks.md (hunk 3 above)
Comment (tpdorsey) [reply docs/concepts/blocks.md:35]: @desertaxle documentation URLs would be great. I mentioned to Evan and Nate including something like a collection_url as well, but I'm inferring this from the documentation URL in my script, so not critical.

File: docs/concepts/blocks.md (hunk 7 above)
Comment (tpdorsey) [reply docs/concepts/blocks.md:54]: Good callout. I'll address this in a future PR.

**Review by desertaxle — APPROVED** [review]

**Pull request merged**